includes/payment-methods/class-wc-stripe-upe-payment-gateway.php

```diff
@@ -153,6 +157,11 @@ public function __construct() {
 				continue;
 			}
 
+			// Show giropay only on the orders page to allow refunds. It was deprecated.
+			if ( WC_Stripe_UPE_Payment_Method_Giropay::class === $payment_method_class && ( ! isset( $query_params['page'] ) || 'wc-orders' !== $query_params['page'] ) ) {
```

Comment:
Hey @wjrosa I just wanted to flag before I get started on testing this that this logic won't catch edit order pages when using legacy WP post order storage. 

So if you go to **WooCommerce > Settings > Advanced > Features (tab)**

<img width="789" alt="Screenshot 2024-07-16 at 1 26 13 PM" src="https://github.com/user-attachments/assets/6c37ec20-1e1f-49d5-8f93-a1d1778528b3">

This option may be disabled, you will need to sync your orders by turning on " compatibility mode" and waiting for it to sync the orders.

Reply:
I also looked into whether it would be possible to use the wp [`get_current_screen()` ](https://developer.wordpress.org/reference/functions/get_current_screen/) here instead of checking the raw GET params, however, at the time we're constructing this class, that function isn't available. :(

Reply:
I'm not sure if you have ideas on how we could fix that, but something like this would work: 

```php
isset( $query_params['post'], $query_params['action'] ) && 'edit' === $query_params['action'] && 'shop_order' === get_post_type( $query_params['post'] )
```

Reply:
> I also looked into whether it would be possible to use the core wp function [get_current_screen() ](https://developer.wordpress.org/reference/functions/get_current_screen/)here instead of checking the raw GET params, however, at the time we're constructing this class, that function isn't available. 😞

Yeah, I tried that as well before reading this comment. What I did was create a simple private method:
```
	private function is_order_details_page() {
		$query_params = wp_unslash( $_GET ); // phpcs:ignore WordPress.Security.NonceVerification.Missing
		if ( wcs_is_custom_order_tables_usage_enabled() ) { // If custom order tables are enabled, we need to check the page query param.
			return isset( $query_params['page'] ) && 'wc-hpos-order-details' === $query_params['page'];
		}
		$is_shop_order_post_type = isset( $query_params['post'] ) && 'shop_order' === get_post_type( $query_params['post'] );
		return isset( $query_params['action'] ) && 'edit' === $query_params['action'] && $is_shop_order_post_type;
	}

```